aws/table_aws_wellarchitected_lens_review_report.go

```diff
@@ -0,0 +1,133 @@
+		Description: "AWS Well-Architected Lens Review Report",
+		List: &plugin.ListConfig{
+			ParentHydrate: listWellArchitectedWorkloads,
+			Hydrate:       getWellArchitectedLensReviewReports,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ResourceNotFoundException"}),
+			},
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "workload_id", Require: plugin.Optional},
+				{Name: "lens_alias", Require: plugin.Optional},
+				{Name: "milestone_number", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: SupportedRegionMatrix(wellarchitectedv1.EndpointsID),
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "lens_alias",
+				Description: "The alias of the lens. For Amazon Web Services official lenses, this is either the lens alias, such as serverless, or the lens ARN, such as arn:aws:wellarchitected:us-east-1:123456789012:lens/my-lens. Each lens is identified by its LensSummary$LensAlias.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "lens_arn",
+				Description: "The ARN for the lens.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "workload_id",
+				Description: "The ID assigned to the workload.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "milestone_number",
+				Description: "The milestone number. A workload can have a maximum of 100 milestones.",
+				Type:        proto.ColumnType_INT,
+			},
+			{
+				Name:        "base64_string",
+				Description: "The Base64-encoded string representation of a lens review report. This data can be used to create a PDF file.",
+				Type:        proto.ColumnType_STRING,
+			},
+		}),
+	}
+}
+
+type ReviewReportInfo struct {
+	Base64String    *string
+	LensAlias       *string
+	LensArn         *string
+	MilestoneNumber int32
+	WorkloadId      *string
+}
+
+//// LIST FUNCTION
+
+func getWellArchitectedLensReviewReports(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	workload := h.Item.(types.WorkloadSummary)
+
+	// Create session
+	svc, err := WellArchitectedClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_wellarchitected_lens_review_report.getWellArchitectedLensReviewReports", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	if d.EqualsQualString("workload_id") != "" {
+		if d.EqualsQualString("workload_id") != *workload.WorkloadId {
+			return nil, nil
+		}
+	}
+	for _, lensAlias := range workload.Lenses {
+		// Check for reduce the numbers of API call if lens_alias or lens_arn is provided in query parameter
+		if d.EqualsQualString("lens_alias") != "" {
+			if d.EqualsQualString("lens_alias") != lensAlias {
+				continue
+			}
+		}
+		input := &wellarchitected.GetLensReviewReportInput{
+			WorkloadId: workload.WorkloadId,
+			LensAlias:  aws.String(lensAlias),
```

Comment:
Does this work correctly for custom lenses?

Reply:
Yes, this will be work fine for the custom lenses also. In the case of custom lens we must have to provide the lens ARN.

The alias of the lens. For Amazon Web Services official lenses, this is either the lens alias, such as serverless, or the lens ARN, such as arn:aws:wellarchitected:us-west-2::lens/serverless. For custom lenses, this is the lens ARN, such as arn:aws:wellarchitected:us-east-1:123456789012:lens/my-lens.